src/Payments/Api.php

```diff
@@ -76,14 +76,25 @@ public function add_payment_method_script_dependencies( $dependencies, $handle )
 		return array_merge( $dependencies, $this->payment_method_registry->get_all_registered_script_handles() );
 	}
 
+	/**
+	 * Returns true if the payment gateway is enabled.
+	 *
+	 * @param object $gateway Payment gateway.
+	 * @return boolean
+	 */
+	private function is_payment_gateway_enabled( $gateway ) {
+		return filter_var( $gateway->enabled, FILTER_VALIDATE_BOOLEAN );
+	}
+
 	/**
 	 * Add payment method data to Asset Registry.
 	 */
 	public function add_payment_method_script_data() {
 		// Enqueue the order of enabled gateways as `paymentGatewaySortOrder`.
 		if ( ! $this->asset_registry->exists( 'paymentGatewaySortOrder' ) ) {
-			$available_gateways = WC()->payment_gateways->payment_gateways();
-			$this->asset_registry->add( 'paymentGatewaySortOrder', array_keys( $available_gateways ) );
+			$payment_gateways = WC()->payment_gateways->payment_gateways();
+			$enabled_gateways = array_filter( $payment_gateways, array( $this, 'is_payment_gateway_enabled' ) );
+			$this->asset_registry->add( 'paymentGatewaySortOrder', array_keys( $enabled_gateways ) );
 		}
```

Comment:
Won't this break the logic that was implemented as a result of [this comment](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2934#discussion_r464924276)?

If you implement the changes I suggested in an earlier comment, there won't be need for this.

Reply:
Not really: enabled and available are different concepts. For example, COD might be enabled but not available depending on the shipping method.

While this code is no longer needed for this PR, what do you think about keeping it? While I see you discussed that was not an information leak, at the same time I don't see any benefit on exposing disabled payment methods in the frontend, so I think we could keep the filter. Thoughts?

Reply:
> For example, COD might be enabled but not available depending on the shipping method.

Ahh, right that makes sense. I'm fine with keeping this but you should make sure testing steps for the work done in #2157 are done to cover any unexpected regressions (don't expect any but still...)

Reply:
I consider testing steps in #2934 already cover any potential regression introduced in this PR. I updated the testing steps in this PR to also point to testing steps in #2934. Do you think we need to add anything else?

Reply:
Nope, if you're happy with that, it works for me!